components/resource_metering/src/model.rs

```diff
@@ -64,10 +65,36 @@ impl Default for RawRecords {
             begin_unix_time_secs: now_unix_time.as_secs(),
             duration: Duration::default(),
             records: HashMap::default(),
+            others: RawRecord::default(),
         }
     }
 }
 
+impl RawRecords {
+    /// Keep a maximum of `k` self.records and aggregate the others into self.others.
+    pub fn keep_top_k(&mut self, k: usize) {
```

Comment:
Seems that we have two `top_k` functions. Is it possible to be one?

Reply:
It depends on whether we can unify the behavior of `keep_top_k(0)`. By the way, why does `Records::keep_top_k(0)` clean up `others` instead of merging all records into `others`? cc @zhongzc 
```rust
         if k == 0 {
             self.records.clear();
             self.others.clear();
             return;
         }
```

Reply:
The per-minute one is being removed. This issue is resolved.

Reply:
This PR should remove the per-minute one.

Reply:
updating